pkg/auth/providers/saml/saml_provider.go

```diff
@@ -176,8 +177,10 @@ func (s *Provider) saveSamlConfig(config *v3.SamlConfig) error {
 	storedSamlConfig.Annotations = config.Annotations
 	config.ObjectMeta = storedSamlConfig.ObjectMeta
 
-	name := fmt.Sprintf("%s:%s-%s", "mgmt-secrets", strings.ToLower(convert.ToString(config.Type)), "spkey")
-	if err := common.CreateOrUpdateSecrets(s.secrets, config.SpKey, "spkey", strings.ToLower(convert.ToString(config.Type))); err != nil {
+	field := strings.ToLower(auth.TypeToField[configType])
+	name := fmt.Sprintf("%s:%s-%s", "mgmt-secrets", strings.ToLower(config.Type), field)
```

Comment:
can we extract the name formatting to its own method? and place it into some shared location as it's used in many places